python/lsst/ts/mtaos/model.py

```diff
@@ -1021,6 +1026,88 @@ def _generate_pipetask_command(
 
         return run_pipetask_cmd
 
+    async def get_image_info(self, visit_id):
+        """Get image information from the butler.
+
+        Parameters
+        ----------
+        visit_id : `int`
+            Visit id of the image.
+
+        Returns
+        -------
+        filter : `str`
+            Filter of the image.
+        rotation_angle : `float`
+            Rotation angle of the image.
+        elevation : `float`
+            Elevation of the image.
+
+        Raises
+        ------
+        ValueError
+            If the visit id is not found in the butler.
+        """
+        butler = dafButler.Butler(self.data_path)
+        refs = butler.query_datasets("raw", where=f"visit={visit_id}")
+
+        if len(refs) == 0:
+            raise ValueError(
+                f"Visit {visit_id} has no associated images in the butler."
+            )
+
+        image = butler.get(refs[0])
+        filter = image.getFilter().bandLabel
+        rotation_angle = image.getMetadata().get("ROTPA")
+        elevation = (
+            image.getMetadata().get("ELSTART") + image.getMetadata().get("ELEND")
+        ) / 2
+
+        return filter, rotation_angle, elevation
+
+    async def get_correction_gain(
+        self, visit_id: int, prev_elevation: float | None
+    ) -> float:
+        """Check if corrections are supposed to be applied.
+
+        Parameters
+        ----------
+        visit_id : `int`
+            Visit id of the image.
+        prev_elevation : `float` or `None`
+            Elevation of the previous image.
+
+        Returns
+        -------
+        gain : `float`
+            Gain to apply to the corrections.
+        """
+        _, _, elevation = await self.get_image_info(visit_id)
+
+        elevation_diff = np.abs(elevation - prev_elevation) if prev_elevation else 0.0
```

Comment:
I think what you are trying to check here is if `prev_elevation` is `None` or not... Please, explicitly check that, e.g.;

```
elevation_diff = np.abs(elevation - prev_elevation) if prev_elevation is not None else 0.0
```